File: source/Deck.java
package source;

import java.util.ArrayList;
import java.util.Random;

public class Deck {

    ArrayList<Integer> cards = new ArrayList<>();

    Random random = new Random();

    public Deck() {

        for (int number = 0; number <= 12; number++) {

            int amount;

            if (number == 0 || number == 1) {
                amount = 1;
            } else {
                amount = number;
            }

            for (int i = 0; i < amount; i++) {

                cards.add(number);
            }
        }
    }

    public int drawCard() {

        int index = random.nextInt(cards.size());

        int card = cards.remove(index);

        return card;
    }
}

File: source/Deck.kt
package source

class Deck {

    private val cards = buildList {
        add(0)
        for (number in 1..12) {
            repeat(number) {
                add(number)
            }
        }
    }.shuffled().toMutableList()

    fun drawCard(): Int {
        return cards.removeAt(cards.lastIndex)
    }
}
